server/src/main/java/org/elasticsearch/action/admin/indices/rollover/TransportRolloverAction.java

```diff
@@ -80,6 +89,24 @@ public TransportRolloverAction(
         this.rolloverService = rolloverService;
         this.client = client;
         this.activeShardsObserver = new ActiveShardsObserver(clusterService, threadPool);
+        this.rolloverTaskExecutor = (currentState, tasks) -> {
+            ClusterStateTaskExecutor.ClusterTasksResult.Builder<RolloverTask> builder = ClusterStateTaskExecutor.ClusterTasksResult
+                .builder();
+            ClusterState state = currentState;
+            for (RolloverTask task : tasks) {
+                try {
+                    state = task.performRollover(state);
+                    builder.success(task);
+                } catch (Exception e) {
+                    builder.failure(task, e);
+                }
+            }
+            String reason = "bulk rollover ["
+                + tasks.stream().map(t -> t.sourceIndex.get() + "->" + t.rolloverIndex.get()).collect(Collectors.joining())
```

Comment:
We gotta limit the output length here. See https://github.com/elastic/elasticsearch/pull/79443 which you can probably reuse here easily.